File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/models/SedTypeTest.java
package no.nav.melosys.eessi.models;

import java.util.stream.Stream;

import org.junit.jupiter.api.Test;

import static no.nav.melosys.eessi.models.SedType.kreverAdresse;
import static org.assertj.core.api.Assertions.assertThat;

class SedTypeTest {

    @Test
    void kreverAdresse_verifiserer_ok() {
        Stream.of(SedType.A001, SedType.A002, SedType.A003, SedType.A004, SedType.A007, SedType.A009, SedType.A010)
            .forEach(sedType -> assertThat(kreverAdresse(sedType))
                .withFailMessage("SedType krever ikke adresse: '%s'", sedType)
                .isTrue());
    }
}

File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/models/SedTypeTest.kt
package no.nav.melosys.eessi.models

import io.kotest.assertions.assertSoftly
import io.kotest.assertions.withClue
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class SedTypeTest {

    @Test
    fun kreverAdresse_verifiserer_ok() {
        val sedTypes = listOf(SedType.A001, SedType.A002, SedType.A003, SedType.A004, SedType.A007, SedType.A009, SedType.A010)
        sedTypes.forEach { sedType ->
            assertSoftly {
                withClue("SedType krever ikke adresse: '$sedType'") {
                    sedType.kreverAdresse() shouldBe true
                }
            }
        }
    }
}
